scripts/Muon/GUI/Common/fitting_widgets/tf_asymmetry_fitting/tf_asymmetry_fitting_model.py

```diff
@@ -0,0 +1,964 @@
+
+    def _update_single_fit_functions_after_sequential(self, workspaces: list, functions: list) -> None:
+        """Updates the single fit functions after a sequential fit has been run on the Sequential fitting tab."""
+        for workspace_index, workspace_name in enumerate(workspaces):
+            if workspace_name in self.dataset_names:
+                dataset_index = self.dataset_names.index(workspace_name)
+                self.single_fit_functions[dataset_index] = functions[workspace_index]
+
+    def _update_tf_asymmetry_single_fit_functions_after_sequential(self, workspaces: list, functions: list) -> None:
+        """Updates the TF single fit functions after a sequential fit has been run on the Sequential fitting tab."""
+        for workspace_index, workspace_name in enumerate(workspaces):
+            if workspace_name in self.dataset_names:
+                dataset_index = self.dataset_names.index(workspace_name)
+                self.update_tf_asymmetry_single_fit_function(dataset_index, functions[workspace_index])
+
+    def _update_simultaneous_fit_function_after_sequential(self, workspaces: list, functions: list) -> None:
+        """Updates the single fit functions after a sequential fit has been run on the Sequential fitting tab."""
+        for fit_index, workspace_names in enumerate(workspaces):
+            if self._are_same_workspaces_as_the_datasets(workspace_names):
+                self.simultaneous_fit_function = functions[fit_index]
+                break
+
+    def _update_tf_asymmetry_simultaneous_fit_function_after_sequential(self, workspaces: list, functions: list) -> None:
+        """Updates the single fit functions after a sequential fit has been run on the Sequential fitting tab."""
+        for fit_index, workspace_names in enumerate(workspaces):
+            if self._are_same_workspaces_as_the_datasets(workspace_names):
+                self.update_tf_asymmetry_simultaneous_fit_function(functions[fit_index])
+                break
+
+    def _update_fit_statuses_and_chi_squared_after_sequential_fit(self, workspaces, fit_statuses, chi_squared_list):
+        """Updates the fit statuses and chi squared after a sequential fit."""
+        if self.simultaneous_fitting_mode:
+            self._update_fit_statuses_and_chi_squared_for_simultaneous_mode(workspaces, fit_statuses, chi_squared_list)
+        else:
+            self._update_fit_statuses_and_chi_squared_for_single_mode(workspaces, fit_statuses, chi_squared_list)
+
+    def _update_fit_statuses_and_chi_squared_for_single_mode(self, workspaces: list, fit_statuses: list,
+                                                             chi_squared_list: list) -> None:
+        """Updates the fit statuses and chi squared after a sequential fit when in single fit mode."""
+        for workspace_index, workspace_name in enumerate(workspaces):
+            if workspace_name in self.dataset_names:
```

Comment:
If this check fails does that mean we've fit to a workspace thats not in our dataset?

Reply:
Sort of, some workspaces shown in the sequential fitting tab are not loaded in the fitting tab and so there is no need to update any information in the fitting tab for specific workspace names. If that makes sense?